### PR TITLE
[dashboard] fix time granularity display values

### PR DESCRIPTION
**Pull request by graceguo-supercat**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [x] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
Time related filter has following fields: time range, time granularity for sql datasource, time granularity for Druid datasource, time column, and Druid time origin. 
- Each of the fields have pre-defined label. 
- For granularity field, their options and display label are associated with datasource. 

Before we only display these predefined labels and values in filter box. Recently we show filter values in more components, like filter indicator, filter scope selector, and these filter fields label shows inconsistently.

This PR is to refactor the time related filter fields's label and display value, so that they should display consistently in all components.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
**before:**
Screen Shot 2019-12-05 at 11 39 58 AM

**after:**
Screen Shot 2019-12-10 at 10 14 27 PM

### TEST PLAN
CI and manual test

### REVIEWERS
@etr2460 @michellethomas @betodealmeida @mistercrunch

### CODE REVIEW

File: superset/assets/src/explore/controls.jsx [hunk 1]
```diff
@@ -814,7 +815,7 @@ export const controls = {
   druid_time_origin: {
     type: 'SelectControl',
     freeForm: true,
-    label: t('Origin'),
+    label: t(TIME_FILTER_LABELS.druid_time_origin),
```
Comment (etr2460) [inline superset/assets/src/explore/controls.jsx]: I think these should no longer be wrapped in `t()`

File: superset/assets/src/explore/constants.js [hunk 2]
```diff
@@ -57,3 +57,11 @@ export const sqlaAutoGeneratedMetricRegex = /^(LONG|DOUBLE|FLOAT)?(SUM|AVG|MAX|M
 export const druidAutoGeneratedMetricRegex = /^(LONG|DOUBLE|FLOAT)?(SUM|MAX|MIN|COUNT)\([A-Z0-9_."]*\)$/i;
 
 export const EXPLORE_ONLY_VIZ_TYPE = ['separator', 'markup'];
+
+export const TIME_FILTER_LABELS = {
+  time_range: 'Time Range',
```
Comment (etr2460) [inline superset/assets/src/explore/constants.js]: I think these should be wrapped in `t()` now

File: superset/assets/src/visualizations/FilterBox/FilterBox.jsx [hunk 3]
```diff
@@ -165,7 +163,7 @@ class FilterBox extends React.Component {
 
   renderDateFilter() {
     const { showDateFilter, chartId } = this.props;
-    const label = t(FILTER_LABELS[TIME_RANGE]);
+    const label = t(TIME_FILTER_LABELS.time_range);
```
Comment (etr2460) [inline superset/assets/src/visualizations/FilterBox/FilterBox.jsx]: i don't think this needs a `t()`

**Review by etr2460 — CHANGES_REQUESTED** [review]
a few changes

File: superset/assets/src/explore/controls.jsx (hunk 1 above)
Comment (graceguo-supercat) [reply superset/assets/src/explore/controls.jsx]: fixed!

File: superset/assets/src/explore/constants.js (hunk 2 above)
Comment (graceguo-supercat) [reply superset/assets/src/explore/constants.js]: fixed!

File: superset/assets/src/visualizations/FilterBox/FilterBox.jsx (hunk 3 above)
Comment (graceguo-supercat) [reply superset/assets/src/visualizations/FilterBox/FilterBox.jsx]: fixed!

**Review by graceguo-supercat — COMMENTED** [review]
At the beginning I feel it's strange to localize `constant`. But in your comment I see it can make less duplicated t() call in multiple places.
Thanks for review!!

**Review by etr2460 — APPROVED** [review]
lgtm!

According to https://github.com/apache-superset/superset-ui/tree/master/packages/superset-ui-translation, only strings should go into the translate functions, so I think this is correct now

**Pull request merged**